drivers/pinctrl/pinctrl_mchp_mec5.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ * Copyright (c) 2021 Linaro Limited
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ * Copyright (c) 2024 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_mec5_pinctrl
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/mchp-xec-pinctrl.h>
+#include <soc.h>
```

Comment:
no soc.h please.

Reply:
@gmarull Is this a new requirement? 

![image](https://github.com/user-attachments/assets/e1fb3157-fce2-426b-8c48-862045c11795)

Reply:
soc.h is never needed, it’s not any “new requirement”. Just make correct use of headers.